src/v/utils/rwlock.h

```diff
@@ -36,8 +39,11 @@ class rwlock_unit {
     rwlock_unit(const rwlock_unit&&) = delete;
     rwlock_unit() = delete;
 
+    explicit operator bool() const noexcept { return _lock != nullptr; }
```

Comment:
i'm still a bit confused about this check. currently there is no way to create a rwlock_unit where _lock is non-null (in which case this check wouldn't be useful), other than move-ctor/assignment. but after you move from something, you generally shouldn't examine it. so what's the use case for this bool operator?

Reply:
it's used in a unit test to check that `target` is initialized in 

```
rwlock_unit target = std::move(source);
BOOST_REQUIRE(target);
```

seastar's semaphore_units [does](https://github.com/scylladb/seastar/blob/6e80e84a3c3cfb2daa5ec0ab37ec264bbc05081f/include/seastar/core/semaphore.hh#L555) it too so I guess it should be fine.